source/databricks/settlement_report/settlement_report_job/domain/report_name_factory.py

```diff
@@ -0,0 +1,88 @@
+from datetime import timedelta
+from zoneinfo import ZoneInfo
+
+from settlement_report_job.domain.market_role import MarketRole
+from settlement_report_job.domain.report_data_type import ReportDataType
+from settlement_report_job.domain.settlement_report_args import SettlementReportArgs
+
+
+class MarketRoleInFileName:
+    """
+    Market role identifiers used in the csv file name in the settlement report.
+    This is not the role of the requesting actor, but the role of the actor that the data is related to.
+    """
+
+    ENERGY_SUPPLIER = "DDQ"
+    GRID_ACCESS_PROVIDER = "DDM"
+
+
+class FileNameFactory:
+    def __init__(self, report_data_type: ReportDataType, args: SettlementReportArgs):
+        self.report_data_type = report_data_type
+        self.args = args
+
+    def create(
+        self,
+        grid_area_code: str,
+        energy_supplier_id: str | None,
+        chunk_index: str | None,
+    ) -> str:
+        if self.report_data_type in {
+            ReportDataType.TimeSeriesHourly,
+            ReportDataType.TimeSeriesQuarterly,
+        }:
+            return self._create_time_series_filename(
+                grid_area_code, energy_supplier_id, chunk_index
+            )
+        else:
+            raise NotImplementedError(
+                f"Report data type {self.report_data_type} is not supported."
+            )
```

Comment:
This pattern is better (exit early and reduce nesting). Now the code is written in the style of guards/pre-conditions.

```suggestion
        if self.report_data_type not in {
            ReportDataType.TimeSeriesHourly,
            ReportDataType.TimeSeriesQuarterly,
        }:
            raise NotImplementedError(
                f"Report data type {self.report_data_type} is not supported."
            )

        return self._create_time_series_filename(
            grid_area_code, energy_supplier_id, chunk_index
        )
```

Reply:
then we will soon have an extra statement. if not in {   } in the beginning and then all the if, elif, elif. I'm not sure that is better?

Reply:
Oh, sorry. Then let's keep it.